superset-frontend/src/views/CRUD/data/dataset/AddDataset/index.tsx

```diff
@@ -73,6 +75,38 @@ export default function AddDataset() {
     Reducer<Partial<DatasetObject> | null, DSReducerActionType>
   >(datasetReducer, null);
   const [hasColumns, setHasColumns] = useState(false);
+  const [linkedDatasets, setLinkedDatasets] = useState<DatasetObject[]>([]);
+  const linkedDatasetNames = linkedDatasets.map(dataset => dataset.table_name);
+  const encodedSchema = dataset?.schema
+    ? encodeURIComponent(dataset?.schema)
+    : undefined;
+
+  const queryParams = dataset?.schema
+    ? rison.encode_uri({
+        filters: [
+          { col: 'schema', opr: 'eq', value: encodedSchema },
+          { col: 'sql', opr: 'dataset_is_null_or_empty', value: '!t' },
+        ],
+      })
+    : undefined;
+
+  const getDatasetsList = () => {
+    SupersetClient.get({
+      endpoint: `/api/v1/dataset/?q=${queryParams}`,
+    })
+      .then(({ json }) => {
+        json.result.map((dataset: any) =>
+          setLinkedDatasets(linkedDatasets => [...linkedDatasets, dataset]),
```

Comment:
I'm not understand this part here.. what's the difference between looping over the json result and adding the datasets to the list as opposed to just setting linkedDatasets to the entire repsonse?

Reply:
When I initially set it to the entire response, it was only setting one dataset at a time for some reason. Looping and adding forces it to make an array of the datasets.

Reply:
When I set the datasets to the entire response like this:
`json.result.map((dataset: any) => setLinkedDatasets(dataset));`
It sets the dataset each time the map runs. So it's always set to one dataset, see this screenshot where I console.logged datasets:

![Screenshot 2022-11-21 at 7 24 39 PM](https://user-images.githubusercontent.com/55605634/203191542-85593934-a05f-442b-a708-3316908c2947.png)

When I originally set this up and ran into this issue setting the datasets into an array, I did some googling for a solution and found this method in [a stackoverflow article](https://stackoverflow.com/questions/54676966/push-method-in-react-hooks-usestate).

Reply:
What about
.then(({ json }) => {
   setDatasets(json.result)
})

Reply:
Are you adding to the existing list of datasets or creating a new list?

Reply:
That worked, thank you! Fixed in [`this commit`](https://github.com/apache/superset/pull/22136/commits/955ffc0b4425550a825757cd9b55f240d0d11b98).

Reply:
nice!